lib/bundle/utils.rb

```diff
@@ -24,6 +24,18 @@ def self.cask_installed?
     @@cask ||= Kernel.system("brew cask --version &>/dev/null")
   end
 
+  def self.installed_formulae
+    @@installed_formulae ||= `brew list -1`.split("\n")
+  end
+
+  def self.outdated_formulae
+    @@outdated_formulae ||= `brew outdated --quiet`.split("\n")
+  end
+
+  def self.pinned_formulae
+    @@pinned_formulae ||= `brew list --pinned`.split("\n")
+  end
```

Comment:
Yes, it has full access to Homebrew API. But as a design choice, we should avoid using it. Part of reason is that Homebrew internal Ruby API is private and may be changed in the future.